sklearn/utils/tests/test_metaestimators.py

```diff
@@ -74,3 +75,33 @@ def test_if_delegate_has_method():
     assert not hasattr(MetaEstTestTuple(HasNoPredict(), HasPredict()), "predict")
     assert not hasattr(MetaEstTestList(HasNoPredict(), HasPredict()), "predict")
     assert hasattr(MetaEstTestList(HasPredict(), HasPredict()), "predict")
+
+
+class AvailableParameterEstimator:
+    """This estimator's `available` parameter toggles the presence of a method
+    """
+
+    def __init__(self, available=True):
+        self.available = available
+
+    @available_if(lambda est: True)
+    def available_func(self):
+        """This is a mock available_if function"""
+        pass
+
+
+def test_available_if_docstring():
+    assert "This is a mock available_if function" in str(
+        AvailableParameterEstimator.__dict__["available_func"].__doc__
+    )
+    assert "This is a mock available_if function" in str(
+        AvailableParameterEstimator.available_func.__doc__
+    )
+    assert "This is a mock available_if function" in str(
+        AvailableParameterEstimator().available_func.__doc__
+    )
+
+
+def test_available_if():
+    assert hasattr(AvailableParameterEstimator(), "available_func")
+    assert hasattr(AvailableParameterEstimator(available=False), "available_func")
```

Comment:
did you mean to have this as 
```suggestion
    assert not hasattr(AvailableParameterEstimator(available=False), "available_func")
```

and make `available_func` only available if `available==True`?

Reply:
Yikes, that's awkward!